src/operation/union/CoverageUnion.cpp

```diff
@@ -80,22 +80,21 @@ std::unique_ptr<Geometry> CoverageUnion::polygonize(const GeometryFactory* gf) {
 
     // Create a vector to manage the lifecycle of a geometry corresponding to each line segment.
     // Polygonizer needs these to stay alive until it does its work.
-    std::unique_ptr<std::vector<std::unique_ptr<Geometry>>> segment_geoms;
-    segment_geoms.reset(new std::vector<std::unique_ptr<Geometry>>());
-    segment_geoms->reserve(segments.size());
+    std::vector<std::unique_ptr<Geometry>> segment_geoms;
+    segment_geoms.reserve(segments.size());
 
     for (const LineSegment& segment : segments) {
         auto seg_geom = segment.toGeometry(*gf);
         p.add(static_cast<Geometry*>(seg_geom.get()));
-        segment_geoms->emplace_back(std::move(seg_geom));
+        segment_geoms.emplace_back(std::move(seg_geom));
     }
 
     if (!p.allInputsFormPolygons()) {
         throw geos::util::TopologyException("CoverageUnion cannot process incorrectly noded inputs.");
     }
 
     auto polygons = p.getPolygons();
-    segment_geoms.reset();
+    segment_geoms.clear();
```

Comment:
I think this line can be removed; the intent of the `reset` was to free the memory early, but that doesn't seem necessary in the current formulation.

Reply:
explicit .clear() now removed